config/default.yaml

```diff
@@ -120,40 +120,49 @@ executor:
         # Launcher container tag to use
         launchVersion: stable
 
-scm:
-    plugin: github
-    github:
-        # The client id used for OAuth with github. Look up GitHub OAuth for details
-        # https://developer.github.com/v3/oauth/
-        oauthClientId: YOU-PROBABLY-WANT-SOMETHING-HERE
-        # The client secret used for OAuth with github
-        oauthClientSecret: AGAIN-SOMETHING-HERE-IS-USEFUL
-        # You can also configure for use with GitHub enterprise
-        # gheHost: github.screwdriver.cd
-        # The username and email used for checkout with github
-        username: sd-buildbot
-        email: dev-null@screwdriver.cd
-        # Secret to add to GitHub webhooks so that we can validate them
-        secret: SUPER-SECRET-SIGNING-THING
-        # Whether it supports private repo: boolean value.
-        # If true, it will ask for read and write access to public and private repos
-        # https://developer.github.com/v3/oauth/#scopes
-        privateRepo: false
-    bitbucket:
-        oauthClientId: YOUR-BITBUCKET-OAUTH-CLIENT-ID
-        oauthClientSecret: YOUR-BITBUCKET-OAUTH-CLIENT-SECRET
-        # The username and email used for checkout with bitbucket
-        # username: sd-buildbot
-        # email: dev-null@screwdriver.cd
-    gitlab:
-        oauthClientId: YOUR-GITLAB-OAUTH-CLIENT-ID
-        oauthClientSecret: YOUR-GITLAB-OAUTH-CLIENT-SECRET
-        # If you have on-premise gitlab, you can specify that here
-        # gitlabHost: mygitlab.com
-        # gitlabProtocol: https
-        # The username and email used for checkout with gitlab
-        # username: sd-buildbot
-        # email: dev-null@screwdriver.cd
+# An example of setting for SCMs is as below.
+# You can set these settings in your local.yaml
+#
+# scms:
+#     - plugin: github
```

Comment:
Sorry, I revert the comment out.But, if the format is an array, the config module seems to override the default setting like as following.So the `scms` settings in default.yaml is just usefull for settings example.
```
> yamlParser.load(fs.readFileSync('./config/default.yaml').toString()).scms;
[ { plugin: 'github',
    config:
     { displayName: 'github',
       oauthClientId: 'YOU-PROBABLY-WANT-SOMETHING-HERE',
       oauthClientSecret: 'AGAIN-SOMETHING-HERE-IS-USEFUL',
       username: 'sd-buildbot',
       email: 'dev-null@screwdriver.cd',
       secret: 'SUPER-SECRET-SIGNING-THING',
       privateRepo: false } } ]
> process.env.SCM_SETTING
'{"displayName":"github","oauthClientId":"clientId","oauthClientSecret":"client-secret"}'
> require('config').get('scms');
{ displayName: 'github',
  oauthClientId: 'clientId',
  oauthClientSecret: 'client-secret' }
```